### ISSUE TITLE
Broken on/off state recognition for NU8042

### ISSUE THREAD

**Issue by fi-sch** [ollo69/ha-samsungtv-smart#28]
**Describe the bug**
v0.3.6 breaks state recognition for my NU8042. I'm not using Smart Things. I can control TV via integration but it does not detect on/off state changes. TV state is always off. 

**Expected behavior**
Behavior from v0.3.5, I've already downgraded to that and it works well.

**Environment details:**
 - Environment: HassOS
 - Home Assistant version installed: 0.112.3
 - Component version installed: v0.3.6
 - Last know working version: v0.3.5
 - TV model: Samsung NU8000 / NU8042 / NU8072

**Output of HA logs**
No errors were seen in HA logs.

**Comment by PDekker** [comment]
Can verify, this is also happening for me on a Q9F . I’m using the SmartThings part of the integration for status report.

**Comment by Pavuucek** [comment]
Also happening with my UE43MU6100 (fw 1280). When I click on a switch button it shows tv in on state and what it does, but reverts to off state in about 5 seconds.

**Comment by ollo69** [comment]
Reading the first post seems not related to SmartThings, that is the only think that I changed in last release, but may be there is a dependencies problem with a library that I changed in the manifest.
Could someone enable debug and attach here log file so that I can check better what is happening? You should add the following lines in configuration.yaml:

```
logger:
  default: info
  logs:
    custom_components.samsungtv_smart: debug
```

Then reboot HA, use TV for a while and then attach here the log file.

Thx

**Comment by fi-sch** [comment]
It looks like it just needed to generate a new token (connection pairing with TV)... I don't know why it did not invoke that pop-up on my TV before. I was running 0.3.6 for more than 24h and launching TV a few times in the meantime. 

I will use an opportunity, to mention the that I've found on both v0.3.5 and v0.3.6. _(However, I don't need it to be fixed as I workaround it easily.)_
There are two logs:
1. From v0.3.5: [home-assistant_v0.3.5.log](https://github.com/ollo69/ha-samsungtv-smart/files/4898331/home-assistant_v0.3.5.log)
I've launched TV at 18.43 (6.43 PM) and let it run till 18.57.
You can also notice that it turned back 'on' at 18.58, less then a minute after I turned it off (via remote). That's a fake state change as TV remained turned off for that time. This fake state lasts around a minute.
2. From v0.3.6: [home-assistant_v0.3.6.log](https://github.com/ollo69/ha-samsungtv-smart/files/4898440/home-assistant_v0.3.6.log)

Same thing here. I've launched TV at 19.10 and let it run till 19.20. Fake state can be seen at 19.31 and it has lasted a little bit longer.

I'm not sure if I sure create separate issue for that. Right now I did a workaround in my automations to ignore this fake state. I'm just checking if TV is unmuted (it is muted while it reports this fake 'on' state) or by comparing last_changed timestamp.

**Comment by ollo69** [comment]
Ok, I will check your log.
It is also interestisting your "check mute state" work-around, I can try to implement it in the component itself. False positive is because in some case TV connect to the network, probably to check for update, while is turned off. But an additional check can solve this issue.
I do not understand how you discovery that new token was required. The popup appear on TV screen or not?

**Comment by fi-sch** [comment]
> False positive is because in some case TV connect to the network, probably to check for update, while is turned off. But an additional check can solve this issue.
I know that it was an issue before. At that time I was checking the source because during the false 'on' state it was reporting "TV/HDMI" – which I don't use ;)  But it was fixed it one of latest releases.
This one, however, appears **only** after I turn off my TV. It triggers only once for about a minute. Then there are no false-posivites for even 24h. So I assume this one is something different.
Anyways, it for sure reports simultaneously "TV/HDMI" source and _is_volume_muted:_ as _true_ in false-positive state.

> I do not understand how you discovery that new token was required. The popup appear on TV screen or not?
Yup. It appeared on TV. Not sure why it worked this time.

**Comment by ollo69** [comment]
I put some additional check and timeout, for this reason false positive are very limited now. There is also an option to change default time-out because different model have different behavior. But is interesting your check, make sense because when TV change from off to on is never muted, also if was muted when turned off (can you confirm that for you is the same?) 
I think I should also add a persistent notification in HA when a new token is requested, because without this can be not clear what is happening (as happen for this issue).

Thx for you contribute.

Please can everybody in this thread confirm that renewing token solve the issue?

**Comment by fi-sch** [comment]
> But is interesting your check, make sense because when TV change from off to on is never muted, also if was muted when turned off (can you confirm that for you is the same?)

Good point. Yup, I've just checked on mine and after being turned on it "unmutes" itself if it was muted before turning off.

> I think I should also add a persistent notification in HA when a new token is requested, because without this can be not clear what is happening (as happen for this issue).

Good idea. That would definitely improve UX, especially for new users.

**Comment by svadel** [comment]
The same behavior with my UE65MU7003. The integration cannot detect on/off state changes. TV state is always off, from the last update.

**Comment by ollo69** [comment]
If possible enable debug log as explained before a post here your log, so that I can check the occuring issue.
As alternative simply try removing authorization for Home-Assistant on your TV, turn it off for a while on turn on again. You should see a popup on the TV requesting for authorization, confirm a check if now it work.

**Comment by svadel** [comment]
Logfile attached:
[log.txt](https://github.com/ollo69/ha-samsungtv-smart/files/4907339/log.txt)

**Comment by ollo69** [comment]
But in all the time frame covered by attached log the TV state was off?

**Comment by svadel** [comment]
Hi!
The TV itself was running.
After restart HA the TV state was OFF and after some trying it turned to
ON, but all my attempts to change volume up or down or mute or the input
source were negative!
Of course, TV has got permission for my HA.
Currently, I running on the previous version of integration and don't have
the problem.

Best Regards,
Vadim

вс, 12 июл. 2020 г. в 01:02, ollo69 <notifications@github.com>:

**Comment by ollo69** [comment]
Can you please try if last release fix the issue and give me your feed-back

**Comment by svadel** [comment]
Thank you, last update has fixed the issue. All works well.

**Comment by fi-sch** [comment]
I will also update that for me all still works good. Except this fake state that occurs only once, around a minute after TV was turned off. – However this was discussed above and it is not the topic of this issue.

Thanks! I'm closing the issue as it looks like it's resolved.
